member.php

```diff
@@ -1599,8 +1600,29 @@
 			$password_length = 8;
 		}
 
-		$password = random_str($password_length);
-		$logindetails = update_password($user['uid'], md5($password), $user['salt']);
+		// Set up user handler.
+		require_once 'inc/datahandlers/user.php';
+		$userhandler = new UserDataHandler('update');
+
+		$userhandler->set_data(array(
+			'uid'		=> $user['uid'],
+			'username'	=> $user['username'],
+			'email'		=> $user['email'],
+			'password'	=> random_str($password_length)
+		));
+
+		if(!$userhandler->validate_user())

```

Comment:
I don't think the whole `validate_user()` needs to run, only `verify_password()` is required because nothing apart from password is changed.